### ISSUE TITLE
What's the correct way to create a `null` value constant expression

### ISSUE THREAD

**Issue by lauxjpn** [dotnet/efcore#23162]
We currently use the following line to create constant expressions that represent `null` values, which works:

```c#
_sqlExpressionFactory.Constant(null, RelationalTypeMapping.NullMapping)
```

However, `ISqlExpressionFactory.Constant()` defines the `value` parameter as `[NotNull]`. Therefore, our usage shown above triggers a warning.

Is there another preferred way to create constant `null` expressions, or should the parameter annotation be changed?

**Comment by smitpatel** [comment]
Given method does not make check, probably annotation is incorrect at this point.

You can avoid the warning by just copying method body as it does not do anything specific processing.
`new SqlConstantExpression(Expression.Constant(value), typeMapping);`

**Comment by smitpatel** [comment]
Hoping annotating APIs with NRT will make such errors caught earlier.

**Comment by lauxjpn** [comment]
No problem, just wanted to make sure I am not misusing the API and missing some other mechanism.

**Comment by smitpatel** [comment]
Fixed in #23185
